xarray/tests/test_missing.py

```diff
@@ -92,7 +92,16 @@ def make_interpolate_example_data(shape, frac_nan, seed=12345, non_uniform=False
 def test_interpolate_pd_compat():
     shapes = [(8, 8), (1, 20), (20, 1), (100, 100)]
     frac_nans = [0, 0.5, 1]
-    methods = ["linear", "nearest", "zero", "slinear", "quadratic", "cubic"]
+    methods = [
+        "linear",
+        "nearest",
+        "zero",
+        "slinear",
+        "quadratic",
+        "cubic",
+        # "next",
+        # "previous",
+    ]
```

Comment:
Ideally we would add `next` and `previous` to the pandas compat tests as well. However, these interpolation methods are not wrapped by pandas yet, so they currently fail.

Reply:
Pandas has this functionality in pad/backfill for `fillna()` rather than in `interpolate`. That could be a good method to try for the tests here.